configuration/configmgr/configmgrlib/mod_templates/IPAddressRangeInput.cpp

```diff
@@ -0,0 +1,164 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "IPAddressRangeInput.hpp"
+#include "TemplateException.hpp"
+#include "Utils.hpp"
+
+void IPAddressRangeInput::setValue(const std::string &range)
+{
+    //
+    // Formats accepted:
+    //  iplist = ip[;ip]
+    //  ip = a.b.c.d | a.b.c.g-h | hostname
+
+    m_values.clear();
+    std::vector<std::string> ipDefs = splitString(range, ";");
+    for (auto &ipDef: ipDefs)
+    {
+        std::string def = trim(ipDef);
+        bool isValid = true;
+
+        //
+        // Figure out what this def entails. Look at the number of parts. If one part, just
+        // save it.
+        std::vector<std::string> addressParts = splitString(def, ".");
+        if (addressParts.size() < 4)
+        {
+            m_values.emplace_back(addressParts[0]);
+        }
+
+        //
+        // If 4 parts, then it must be an ip address or range
+        else if (addressParts.size() == 4)
+        {
+            std::string ipAdressBase;
+            //
+            // check to make sure the first 3 parts are valid numbers
+            for (unsigned i=0; i<3 && isValid; ++i)
+            {
+                try
+                {
+                    int num = std::stoi(addressParts[i]);
+                    if (num <= 0 || num > 255)
+                    {
+                        isValid = false;
+                    }
+                    else
+                    {
+                        ipAdressBase += addressParts[i] + ".";
+                    }
+                }
+                catch (...)
+                {
+                    isValid = false;
+                }
+            }
+
+            //
+            // If valid, look at the last part, either a valid octet of the address or a-b range
+            if (isValid)
+            {
+                int start, stop;
+                if (addressParts[3].find('*') != std::string::npos)
+                {
+                    std::vector<std::string> rangeParts = splitString(addressParts[3], "*");
+                    try
+                    {
+                        start = std::stoi(rangeParts[0]);
+                        int num = std::stoi(rangeParts[1]);
```

Comment:
Does num need to be checked for a valid range? IE num > 0 && start+num <= 255

Reply:
Yes, in hind sight it does. It's a count, but I better make sure it doesn't overflow.